integreat_cms/cms/migrations/0079_alter_page_mirrored_page.py

```diff
@@ -0,0 +1,61 @@
+# Generated by Django 3.2.20 on 2023-09-07 10:59
+
+from django.db import migrations
+
+from ..constants import region_status
+
+
+def update_implicitly_archived_pages(apps, schema_editor, pages):
+    """
+    Set the field mirrored_pages to None if implicitly archived
+
+    :param apps: The configuration of installed applications
+    :type apps: ~django.apps.registry.Apps
+
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
+
+    :param pages: The pages that are archived
+    :type pages: ~django.db.models.query.QuerySet [ ~integreat_cms.cms.models.Page ]
+    """
+
+    Page = apps.get_model("cms", "Page")
+    if child_pages := Page.objects.filter(parent__in=pages):
+        child_pages.update(mirrored_page=None)
+        update_implicitly_archived_pages(apps, schema_editor, child_pages)
+
+
+def update_mirrored_pages(apps, schema_editor):
+    """
+    Set the field mirrored_page to None when a page is archived or belongs to an archived region.
+
+    :param apps: The configuration of installed applications
+    :type apps: ~django.apps.registry.Apps
+
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
+    """
+
+    Page = apps.get_model("cms", "Page")
+
+    pages_archived_region = Page.objects.filter(region__status=region_status.ARCHIVED)
+    pages_archived_region.update(mirrored_page=None)
+
+    pages_explicitely_archived = Page.objects.filter(explicitly_archived=True)
+    pages_explicitely_archived.update(mirrored_page=None)
+
+    update_implicitly_archived_pages(apps, schema_editor, pages_explicitely_archived)
```

Comment:
```suggestion

    pages_archived = [
        Page.objects.filter(
            tree_id=page.tree_id, lft__range=(page.lft, page.rgt - 1)
        ).order_by()
        for page in Page.objects.filter(explicitly_archived=True)
    ]

    page_ids = pages_archived_region.union(*pages_archived).values_list("id", flat=True)

    Page.objects.filter(id__in=page_ids).update(mirrored_page=None)
```
I think this can be optimized a bit.
We don't have access to the page model's methods like `get_tree()`, but we can use the [queryset filter in its source definition](https://github.com/django-treebeard/django-treebeard/blob/f9674d0adf2ae3077c4475895f0df074179102c7/treebeard/ns_tree.py#L627-L629).